app/src/marketplace/containers/EditProductPage/EditorNav.jsx

```diff
@@ -155,12 +163,32 @@ const EditorNav = () => {
         }
     }, [clickTargetRef])
 
+    const activeSectionRef = useRef()
+    activeSectionRef.current = activeSectionId
+
+    useEffect(() => {
+        if (lastSectionRef.current) {
+            scrollTo(lastSectionRef.current, false)
+            setActiveSectionId((prevActive) => {
+                if (prevActive !== lastSectionRef.current) {
+                    return lastSectionRef.current
+                }
+                return prevActive
+            })
```

Comment:
It seems to be a `if something then B else B` case, see:
```js
setActiveSectionId((prevActive) => {
    if (prevActive !== lastSectionRef.current) {
        return lastSectionRef.current
    }
    // At this point we know that prevActive and lastSectionRef.current are the same.
    // It doesn't matter which one we return.
    return prevActive // or lastSectionRef.current
})
```
If I'm right, then `setActiveSectionId(lastSectionRef.current)` should be enough.

Reply:
Your logic is indeed sound :)